<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="object" default="door"/>
    <arg name="tf_prefix" default=""/>

<!--    <arg name="description_file" default="$(find mppi_manipulation)/data/panda_standalone.urdf.xacro"-->
<!--         if="$(arg fixed_base)"/>-->
<!--    <arg name="description_file" default="$(find mppi_manipulation)/data/panda_mobile.urdf.xacro"-->
<!--         unless="$(arg fixed_base)"/>-->

<!--    <arg name="object_height" default="0.0" if="$(arg fixed_base)"/>-->
<!--    <arg name="object_height" default="0.45" unless="$(arg fixed_base)"/>-->

    <include file="$(find mppi_manipulation)/launch/load.launch"/>
<!--    <param name="robot_description" command="$(find xacro)/xacro $(arg description_file)-->
<!--      resources_prefix:='file://'-->
<!--      collision_links:=false-->
<!--      collision_hand:=true-->
<!--      add_finger:=true"/>-->

<!--    <param name="robot_description_raisim" command="$(find xacro)/xacro $(arg description_file)-->
<!--      collision_links:=false-->
<!--      collision_hand:=true-->
<!--      collision_finger:=true-->
<!--      collision_format:='obj'-->
<!--      add_finger:=true-->
<!--      no_position_limits:=true"/>-->

<!--    <param name="object_description"-->
<!--           command="$(find xacro)/xacro $(find mppi_manipulation)/data/objects/$(arg object).urdf.xacro-->
<!--           resource_prefix:='file://'-->
<!--           height:=$(arg object_height)"/>-->
<!--    <param name="object_description_raisim"-->
<!--           command="$(find xacro)/xacro $(find mppi_manipulation)/data/objects/$(arg object).urdf.xacro height:=$(arg object_height)"/>-->


    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="100"/>
        <param name="use_tf_static" value="true"/>
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
        <remap from="robot_description" to="robot_description"/>
        <remap from="joint_states" to="/joint_states"/>
    </node>

    <node name="object_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" output="screen">
        <param name="use_tf_static" value="true"/>
        <param name="publish_frequency" value="100"/>
        <remap from="robot_description" to="object_description"/>
        <remap from="joint_states" to="/object/joint_state"/>
    </node>

    <node name="manipulation_raisim" type="manipulation_raisim" pkg="mppi_manipulation" output="screen"
          launch-prefix="gdb -ex run --args">
      <rosparam command="load" file="$(find mppi_manipulation)/params/params.yaml"/>
      <param name="activation_file" value="/home/boyang/raisim_ws/raisimLib/build/examples/rsc/activation.raisim"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mppi_manipulation)/config/rviz/config.rviz" output="screen"/>
</launch>
